proxy/agent.go

```diff
@@ -182,6 +188,12 @@ func (a *agent) Run(stop <-chan struct{}) {
 			}
 
 		case status := <-a.statusCh:
+			// delete epoch record and update current config
+			// avoid self-aborting on non-abort error
+			delete(a.epochs, status.epoch)
```

Comment:
BTW - may be worth a comment that 'all changes to a.epocks and a.abortCh happen in the Run thread, so they don't require a mutex' ( in case anyone wonders )